ArgoProject/src/systems/CollisionSystem.cpp

```diff
@@ -362,4 +364,24 @@ void app::sys::CollisionSystem::playerHazardCollisions()
 			}
 		});
 	});
+}
+
+void app::sys::CollisionSystem::playerGoalCollisions()
+{
+	m_registry.view<comp::Collision, comp::Input, comp::Location, comp::Dimensions>(entt::persistent_t())
+		.each([&, this](app::Entity const entity, comp::Collision & collision, comp::Input & input, comp::Location & location, comp::Dimensions & dimensions)
+	{
+		m_registry.view<comp::Collision, comp::Goal>()
+			.each([&, this](app::Entity const secEntity, comp::Collision & secCollision, comp::Goal & goal)
+		{
+			if (entity != secEntity)
+			{
+				auto const & collisionCheck = app::vis::CollisionBoundsBoolVisitor::collisionBetween(collision.bounds, secCollision.bounds);
+				if (collisionCheck)
```

Comment:
```suggestion
				if (app::vis::CollisionBoundsBoolVisitor::collisionBetween(collision.bounds, secCollision.bounds))
```
No need to store the temporary variable `auto const & collisionCheck` as it is a boolean which is evaluated by the if statement and never used anywhere else.